### ISSUE TITLE
`skip_ansible_lint` tag ignored by ansible-lint 6.5.0

### ISSUE THREAD

**Issue by foutrelis** [ansible/ansible-lint#2320]
##### Summary

Adding the `skip_ansible_lint` tag to a task does not seem to exclude it from linting. This used to work in previous versions of ansible-lint (up to 6.4.0).

##### Issue Type

- Bug Report

##### Ansible and Ansible Lint details

```
$ ansible --version
ansible [core 2.13.3]
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/home/user/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python3.10/site-packages/ansible
  ansible collection location = /home/user/.ansible/collections:/usr/share/ansible/collections
  executable location = /usr/bin/ansible
  python version = 3.10.6 (main, Aug  3 2022, 17:39:45) [GCC 12.1.1 20220730]
  jinja version = 3.1.2
  libyaml = True

$ ansible-lint --version
ansible-lint unknown using ansible 2.13.3

$ pacman -Q ansible-lint
ansible-lint 6.5.0-2
```

- ansible installation method: OS package
- ansible-lint installation method: OS package

##### OS / ENVIRONMENT

Arch Linux x86_64

##### STEPS TO REPRODUCE

```yaml
- hosts: localhost
  tasks:
  - name: Foo
    command: foo
    tags:
      - skip_ansible_lint
```

##### Desired Behavior

No errors and/or warnings.

##### Actual Behavior

```
WARNING  Listing 2 violation(s) that are fatal
fqcn-builtins: Use FQCN for builtin actions.
skip.yml:3 Task/Handler: Foo

no-changed-when: Commands should not change things if nothing needs doing.
skip.yml:3 Task/Handler: Foo

You can skip specific rules or tags by adding them to your configuration file:
# .config/ansible-lint.yml
warn_list:  # or 'skip_list' to silence them completely
  - fqcn-builtins  # Use FQCN for builtin actions.
  - no-changed-when  # Commands should not change things if nothing needs doing.

Finished with 2 failure(s), 0 warning(s) on 1 files.
```

**Comment by foutrelis** [comment]
Bisects to b765dab16c980c55a8366226dd484fb9c6d183a9.

**Comment by ssbarnea** [comment]
@foutrelis Assigning it to me because I also happen to have another change in progress for fixing other issues related to skips, including this one. Hopefully I get it fixed today. There is even an PR open for it but that is outdated, so no need to look at it, yet.
